native/cocos/engine/Engine.cpp

```diff
@@ -117,8 +119,9 @@ int32_t Engine::init() {
 #if CC_USE_PROFILER
     _profiler = ccnew Profiler();
 #endif
-
-    _scriptEngine = ccnew se::ScriptEngine();
+    if (!_scriptEngine) {
+        _scriptEngine = ccnew se::ScriptEngine();
+    }
```

Comment:
Please add comments why need to do it like this.

Reply:
OK

Reply:
> The script engine is created in the engine constructor for the first time.

Please add a reason for it.

Reply:
Could users's custom JSB registeration in Game.cpp be invoked again while restarting?